### DISCUSSION TITLE
martinize2: command not found

### DISCUSSION THREAD

**Discussion by mangochiral** [marrink-lab/vermouth-martinize#464] [Q&A] [answered]
vermoth is installed 
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: vermouth in /home/chandrima/.local/lib/python3.10/site-packages (0.7.3)
Requirement already satisfied: numpy in /home/chandrima/.local/lib/python3.10/site-packages (from vermouth) (1.23.2)
Requirement already satisfied: pbr in /home/chandrima/.local/lib/python3.10/site-packages (from vermouth) (5.10.0)
Requirement already satisfied: networkx~=2.0 in /home/chandrima/.local/lib/python3.10/site-packages (from vermouth) (2.8.6)

**Comment by fgrunewald** [comment]
@mangochiral we need some more information in order to figure out what is going on:

- operating system
- did you do a pip or conda install
- have you sourced the virtual environment?

**Comment by mangochiral** [comment]
its Ubuntu 22.04
martinize2 is ../.local/bin
sourcing is from there using full file path is work but now there is DSSP (4.0..)error.
Traceback (most recent call last):
  File "/home/chandrima/.local/bin/martinize2", line 797, in <module>
    entry()
  File "/home/chandrima/.local/bin/martinize2", line 640, in entry
    AnnotateDSSP(executable=args.dssp, savedir='.').run_system(system)
  File "/home/chandrima/.local/lib/python3.10/site-packages/vermouth/processors/processor.py", line 37, in run_system
    mols.append(self.run_molecule(molecule))
  File "/home/chandrima/.local/lib/python3.10/site-packages/vermouth/dssp/dssp.py", line 462, in run_molecule
    annotate_dssp(molecule, self.executable, self.savedir)
  File "/home/chandrima/.local/lib/python3.10/site-packages/vermouth/dssp/dssp.py", line 293, in annotate_dssp
    secstructs = run_dssp(system, executable, savefile)
  File "/home/chandrima/.local/lib/python3.10/site-packages/vermouth/dssp/dssp.py", line 216, in run_dssp
    raise DSSPError(message.format(err=process.stderr, file=tmpfile_name))
vermouth.dssp.dssp.DSSPError: DSSP encountered an error. The message was Error parsing PDB at line 1
Error trying to load file "/home/chandrima/Documents/dssp_in_lsydnud_.pdb"
Expected record CRYST1 but found ATOM  
. The input file provided to DSSP can be found at /home/chandrima/Documents/dssp_in_lsydnud_.pdb.

**Reply by mangochiral** [reply]
Also I have used the CRYST1 line still not working

**Comment by pckroon** [comment, marked as answer]
See https://www.baeldung.com/linux/path-variable#:~:text=The%20PATH%20variable%20is%20an,path%20when%20running%20a%20command. (or google) about PATH variables and why the martinize2 works with the full path, but not as is.

See issue #364 about DSSP versions

**Reply by mangochiral** [reply]
Hi pckroon,
I am non tech person
I am trying with different dssp 3.1.4. I am unable to locate liboost 1.54. Can help me in troubleshooting this?
The message was /usr/local/bin/mkdssp: error while loading shared libraries: libboost_iostreams.so.1.54.0: cannot open shared object file: No such file or directory
I did all this 
Compiler:
* Must support at least the c++ 11 standard.

System libraries:

* libboost version >= 1.48
* libz
* libbz2
* autoconf
* automake
* autotools-dev

### Instructions

Download and uncompress the dssp [source code archive][2] (version >= 2.2.6):

    wget https://github.com/cmbi/dssp/archive/dssp-?.?.?.tar.gz
    tar -zxvf dssp-?.?.?.tar.gz
    cd dssp-?.?.?.tar.gz

Configure and build the dssp executables:

    ./autogen.sh
    ./configure
    make

To build only one executable of the dssp project, e.g. `mkdssp`, type:

    make mkdssp

To test the `mkdssp` executable type:

    ./mkdssp

To add the executables to /usr/local/bin type:

    sudo make install

./mkdssp: error while loading shared libraries: libboost_iostreams.so.1.54.0: cannot open shared object file: No such file or directory

**Reply by pckroon** [reply]
> Can help me in troubleshooting this?

No, sorry. I don't have time available.

**Reply by mangochiral** [reply]
Any recommendation where I can find help

**Reply by mangochiral** [reply]
Thank you I have resolved all the problem. Installed dssp 3.1.2, and used sudo pip vermouth to install globally
